src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/events/ActionPlanAppointmentEventPublisher.kt

```diff
@@ -0,0 +1,49 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.events
+
+import org.springframework.context.ApplicationEvent
+import org.springframework.context.ApplicationEventPublisher
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.component.LocationMapper
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.controller.ActionPlanSessionController
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.ActionPlanSession
+
+enum class ActionPlanAppointmentEventType {
+  ATTENDANCE_RECORDED,
+  BEHAVIOUR_RECORDED,
+  SESSION_FEEDBACK_RECORDED,
+}
+
+class ActionPlanAppointmentEvent(source: Any, val type: ActionPlanAppointmentEventType, val actionPlanSession: ActionPlanSession, val detailUrl: String, val notifyPP: Boolean) : ApplicationEvent(source) {
+  override fun toString(): String {
+    return "ActionPlanAppointmentEvent(type=$type, appointment=${actionPlanSession.id}, detailUrl='$detailUrl', source=$source)"
```

Comment:
In the domain, people mention delivery appointments -- could we rename these to match the domain language, please?